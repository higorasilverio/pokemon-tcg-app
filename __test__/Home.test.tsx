import { screen } from '@testing-library/dom'
import '@testing-library/jest-dom'
import React from 'react'
import Home from '../src/pages/index'
import { renderWithProviders } from '../src/utils/renderWithProvider'

jest.mock('next/router', () => ({
  useRouter() {
    return {
      locale: 'pt-BR'
    }
  }
}))

describe('Home', () => {
  it('renders homepage unchanged', () => {
    const { container } = renderWithProviders(<Home />)
    expect(container).toMatchSnapshot()
  })

  it('renders a heading', () => {
    renderWithProviders(<Home />)

    const heading = screen.getByText(/pokÃ©mon tcg app/i)

    expect(heading).toBeInTheDocument()
  })

  it('renders location correctly', () => {
    const result = renderWithProviders(<Home />)
    const location = result.container.querySelector('p')

    expect(location).toHaveTextContent('Location: pt-BR')
  })
})
