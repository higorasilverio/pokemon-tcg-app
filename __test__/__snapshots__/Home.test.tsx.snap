// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <h1>
    Pok√©mon TCG APP
  </h1>
  <p>
    Location: 
    pt-BR
  </p>
  <div>
    <button
      type="button"
    >
      Toggle location
    </button>
  </div>
  <div
    style="margin-top: 25px;"
  >
    <a
      href="/about"
    >
      About
    </a>
  </div>
  <div
    style="margin-top: 25px;"
  >
    <button
      type="button"
    >
      Toggle state
    </button>
  </div>
</div>
`;
